### ISSUE TITLE
rrule.after returns null after being passed a valid Date object

### ISSUE THREAD

**Issue by dopatraman** [jkbrzt/rrule#640]
```javascript
const { RRule, datetime } = require('rrule');
const rRule = RRule.fromString("DTSTART:20240711T000000Z\nRRULE:FREQ=YEARLY;INTERVAL=1;COUNT=1")

// Given this date:
const dd = new Date('2024-07-11 00:00:00'); // 2024-07-11T07:00:00.000Z

// and this rrule datetime
const dt = datetime(dd.getYear(), dd.getMonth(), dd.getDay()); // 2024-07-11T07:00:00.000Z

```
If I call `rRule.after(dt)` i get `null`:

```javascript
rRule.after(dt) // null
```

Why? As far as I can tell this is how the function is meant to be used. The expected output would be 1 year into the future, based on the recurrence string.

RRULE version: 2.8.1
OS: MacOS Sonoma
Timezone: Tue Oct  1 15:24:16 PDT 2024

### Reporting an issue

Thank you for taking an interest in `rrule`! Please include the following in
your report:

- [x] Verify that you've looked through existing issues for duplicates before
      creating a new one
- [x] Code sample reproducing the issue. Be sure to include all input values you
      are using such as the exact RRule string and dates.
- [x] Expected output
- [x] Actual output
- [x] The version of `rrule` you are using
- [x] Your operating system
- [x] Your local timezone (run `$ date` from the command line
      of the machine showing the bug)

**Comment by RohovDmytro** [comment]
@dopatraman have you figured out the deal?

**Comment by ggaabe** [comment]
Hi, this library isn't maintained anymore. A new one has been built following the original maintainer's suggestions, with the full RFC 5545 implementation:

https://github.com/ggaabe/rrule-temporal

**Comment by RohovDmytro** [comment]
> Hi, this library isn't maintained anymore. A new one has been built following the original maintainer's suggestions, with the full RFC 5545 implementation:
> 
> https://github.com/ggaabe/rrule-temporal

Can you please ellaborate on this project more, do you have any experience with it and how production tested it?

**Comment by ggaabe** [comment]
Yeah, I'm using it in production now. It has over 700 unit tests that you can take a look at, all edge cases that we know of are covered. It was built to fully follow the RFC spec based on the Temporal object that the maintainers of this project recommended; the purpose was so it could work easily across all timezones
